lib/hsa/mcwamp_hsa.cpp

```diff
@@ -4176,14 +4176,14 @@ void HSAQueue::dispose() {
 
         Kalmar::HSADevice* device = static_cast<Kalmar::HSADevice*>(getDev());
 
+        wait();
+
```

Comment:
Some comment would help. It seems calling 'wait()' is equivalent to holding `qmutex` and `wait_no_lock()` which is what is lines 4185 below after locking `rocrQueuesMutex`. Does this fix imply that `rocrQueuesMutex` should NOT be held before `qmutex` as it may cause deadlock? If so, shouldn't line 4185 locking of `qmutex` be removed?
Or, could moving acquiring of `rocrQueuesMutex` to just before calling `removeRocrQueue()` help?